src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -0,0 +1,98 @@
+import ITechRecordWrapper from "../../../@Types/ITechRecordWrapper";
+import HTTPError from "../../models/HTTPError";
+import {ERRORS, RECORD_COMPLETENESS_ENUM, VEHICLE_TYPE} from "../../assets/Enums";
+import {
+  carCoreMandatorySchema,
+  hgvCoreMandatorySchema, lgvCoreMandatorySchema, motorcycleCoreMandatorySchema,
+  psvCoreMandatorySchema,
+  psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, trlCoreMandatorySchema,
+  trlCoreMandatoryVehicleAttributes
+} from "./CoreMandatoryValidations";
+import {ObjectSchema} from "@hapi/joi";
+import {
+  hgvNonCoreMandatorySchema,
+  psvNonCoreMandatorySchema,
+  trlNonCoreMandatorySchema
+} from "./NonCoreMandatoryValidations";
+
+const validateRecordCompleteness = (validationSchema: ObjectSchema | undefined, techRecordFields: any) => {
+  if (validationSchema) {
+    return validationSchema.validate(techRecordFields, {stripUnknown: true});
+  } else {
+    return undefined;
+  }
+};
+
+const validateVehicleAttributes = (vehicleType: string, vehicleAttributes: any) => {
+  if (vehicleType !== VEHICLE_TYPE.TRL) {
+    return validateRecordCompleteness(psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  } else {
+    return validateRecordCompleteness(trlCoreMandatoryVehicleAttributes, vehicleAttributes);
+  }
+};
+
+const validateCoreAndNonCoreMandatoryTechRecordAttributes = (vehicleType: string, techRecordWrapper: ITechRecordWrapper) => {
+  let coreMandatoryValidationResult;
+  let nonCoreMandatoryValidationResult;
+  let coreMandatorySchema: ObjectSchema;
+  let nonCoreMandatorySchema: ObjectSchema | undefined;
+  let techRecord = techRecordWrapper.techRecord[0];
+  if (vehicleType === VEHICLE_TYPE.HGV) {
+    coreMandatorySchema = hgvCoreMandatorySchema;
+    nonCoreMandatorySchema = hgvNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.PSV) {
+    coreMandatorySchema = psvCoreMandatorySchema;
+    nonCoreMandatorySchema = psvNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.TRL) {
+    techRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm} as any;
```

Comment:
because the validation schema for the non-core mandatory attributes for TRL is different from the interface we have. PrimaryVrm is a non-core mandatory attribute for TRL